firestore.rules

```diff
@@ -1,9 +1,32 @@
 rules_version = '2';
+
 service cloud.firestore {
   match /databases/{database}/documents {
-    match /{document=**} {
+    match /users/{userId} {
+      allow read, write: if request.auth.uid == userId;
+    }
+    match /stories/{storyId} {
       allow read;
-      allow write: if request.auth != null;
+      // TODO: check for required fields
+      // https://firebase.google.com/docs/firestore/security/rules-fields#requiring_specific_fields_in_new_documents
+      allow create: if request.auth != null;
+      allow update: if request.auth != null && request.auth.uid == resource.data.metaData.authorId;
+      allow delete: if request.auth != null && request.auth.uid == resource.data.metaData.authorId;
+
+      match /translations/{translationId} {
+        allow read;
+        allow create: if request.auth != null && request.auth.uid == request.resource.data.metaData.authorId;
+        allow update: if request.auth != null && request.auth.uid == resource.data.metaData.authorId;
+        allow delete: if request.auth != null && request.auth.uid == resource.data.metaData.authorId;
+
+        // Not sure yet how exactly to protect these two subcollections.
+        match /assets/{assetId} {
```

Comment:
We can do it later by checking if the request is by either translation author or the asset author. We will need to keep more metadata on each asset, such as author and likes.

Reply:
Yes, I thought about using the parent translation document, but also realized it may not be that important for now, or might even be too restrictive (since other users can suggest edits to the translation), so I decided this was good enough (at least better than complete freedom to edit everything, as it was 😅)